lib/system/cellseqs_v2.nim

```diff
@@ -16,20 +16,21 @@ type
     len, cap: int
     d: CellArray[T]
 
-proc add[T](s: var CellSeq[T], c: T; t: PNimTypeV2) {.inline.} =
+proc resize[T](s: var CellSeq[T]) =
+  s.cap = s.cap * 3 div 2
+  var newSize = s.cap * sizeof(CellTuple[T])
+
+  when compileOption("threads"):
+    let oldData = s.d
+    s.d = cast[CellArray[T]](allocShared(newSize))
```

Comment:
We have `reallocShared` too.

Reply:
Oooo I don't know why I didn't see that before! The code looks _way_ better now. Addressed here: 9fdb2f06f1e78b2af77479780f432c2c6ab07b4f